### PR TITLE
seperate input queue for ui

### PR DESCRIPTION
**Pull request by GarmOfGnipahellir**
fix for #244 
this behavior might be undesirable but I needed a quick fix and decided to share

### CODE REVIEW

File: src/input.rs [hunk 1]
```diff
@@ -125,6 +125,12 @@ pub fn get_char_pressed() -> Option<char> {
     context.chars_pressed_queue.pop()
 }
 
+pub fn get_char_pressed_ui() -> Option<char> {
```
Comment (not-fl3) [inline src/input.rs:128]: Could you make it `pub(crate)`? 
UI will consume the entire queue anyway, so no need to make this function available in the public API

**Comment by not-fl3** [comment]
Thanks for PR!

**Comment by GarmOfGnipahellir** [comment]
No problem!

**Comment by oofsauce** [comment]
has this been pushed to crates.io yet? i'm still hitting the bug on 0.3.8

**Pull request merged**